elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala

```diff
@@ -28,12 +32,49 @@ trait MappingHandlers {
       }
     }
 
-
     override def build(request: GetMappingRequest): ElasticRequest = {
       val endpoint = request.indexes match {
         case Indexes(Nil)       => "/_mapping"
         case Indexes(indexes)   => s"/${indexes.mkString(",")}/_mapping"
       }
+
+      ElasticRequest("GET", endpoint)
+    }
+  }
+
+  implicit object GetFieldMappingRequest extends Handler[GetFieldMappingRequest, Seq[IndexFieldMapping]] {
+    override def responseHandler: ResponseHandler[Seq[IndexFieldMapping]] = new ResponseHandler[Seq[IndexFieldMapping]] {
+
+      override def handle(response: HttpResponse) : Either[ElasticError, Seq[IndexFieldMapping]] = response.statusCode match {
+        case 201 | 200       =>
+          val raw = ResponseHandler.fromResponse[Map[String, Map[String, Map[String, Map[String, Any]]]]](response)
+          Right(raw.map {
+            case (index, types) =>
+              val mappings = types("mappings").map {
+                case (_, mapping) => FieldMapping(mapping("full_name").toString, mapping("mapping").asInstanceOf[Map[String, Any]])
+              }.toSeq
+              IndexFieldMapping(index, mappings)
+          }.toSeq)
+        case _              =>
+          try {
+            Left(ElasticError.parse(response))
+          }catch{
+            case _ : Throwable => sys.error(s"""Failed to parse error response: \n${response.toString}""")
+          }
+      }
+    }
+
+    override def build(request: GetFieldMappingRequest): ElasticRequest = {
+      val baseEndpoint = request.indexes match {
+        case Indexes(Nil)       => "/_mapping"
+        case Indexes(indexes)   => s"/${indexes.mkString(",")}/_mapping"
+      }
+
+      val endpoint = request.fields match {
+        case Nil => throw new IllegalArgumentException("Empty fields in GetFieldMappingRequest. Use GetMappingRequest instead.")
```

Comment:
Perhaps this isn't the right thing to do? A field mapping request requires at least one field;.